pkg/proxy/apis/config/validation/validation.go

```diff
@@ -67,8 +68,17 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if _, _, err := net.ParseCIDR(config.ClusterCIDR); err != nil {
-			allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), config.ClusterCIDR, "must be a valid CIDR block (e.g. 10.100.0.0/16)"))
+		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
+			cidrs := strings.Split(config.ClusterCIDR, ",")
+			for _, cidr := range cidrs {
+				if _, _, err := net.ParseCIDR(cidr); err != nil {
+					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
+				}
+			}
+		} else {
```

Comment:
so technically here the `if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {` branch is not needed, until we add more logic related to the feature gate.

even for a single entry such as `config.ClusterCIDR = 10.100.0.0/16`, by using `cidrs := strings.Split(config.ClusterCIDR, ",")` and the `for _, cidr := range cidrs {` loop, single values will be handled too.

but i guess i'm ok leaving the separate branches for visibility of the new feature.